vamb/__main__.py

```diff
@@ -939,6 +980,306 @@ def run(
+    MODEL_PATH = vamb_options.out_dir.joinpath("predictor_model.pt")
+    with open(MODEL_PATH, 'wb') as modelfile:
+        model.trainmodel(
+            dataloader_joint,
+            nepochs=predictor_training_options.nepochs,
+            modelfile=modelfile,
+            logfile=logfile,
+            batchsteps=predictor_training_options.batchsteps,
+        )
+    log(f"Finished training the taxonomy predictor in {round(time.time() - predictortime, 2)} seconds.", logfile, 0)
+    predicted_vector = model.predict(dataloader_vamb)
+
+    log(f"Writing the taxonomy predictions", logfile, 0)
+    df_gt = pd.DataFrame({'contigs': names})
+    nodes_ar = np.array(nodes)
+    predictions = []
+    for i in range(len(df_gt)):
+        predictions.append(';'.join(nodes_ar[predicted_vector[i] > 0.5][1:]))
+    df_gt[f'predictions'] = predictions
+
+    df_mmseq_sp = df_mmseq[(df_mmseq[2] == 'species')]
+    mmseq_map = {k: v for k, v in zip(df_mmseq_sp[0], df_mmseq_sp[8])}
+    preds = []
+    for i, r in df_gt.iterrows():
+        pred_line = r[f'predictions'].split(';')
+        try:
+            mmseq_line = mmseq_map.get(r['contigs'], '').split(';')
+        except AttributeError:
+            preds.append(';'.join(pred_line))
+            continue
+        if mmseq_line[0] != '':
+            for i in range(len(mmseq_line)):
+                if i < len(pred_line):
+                    pred_line[i] = mmseq_line[i]
+        preds.append(';'.join(pred_line))
+    df_gt[f'predictions'] = preds
+
+    PREDICTED_PATH = vamb_options.out_dir.joinpath("results_taxonomy_predictor.csv")
+    df_gt.to_csv(PREDICTED_PATH, index=None)
+
+    log(f"\nCompleted taxonomy predictions in {round(time.time() - begintime, 2)} seconds.", logfile, 0)
+
+
+def run_vaevae(
+    vamb_options: VambOptions,
+    comp_options: CompositionOptions,
+    abundance_options: AbundanceOptions,
+    vae_training_options: VAETrainingOptions,
+    predictor_training_options: VAETrainingOptions,
+    taxonomy_options: TaxonomyOptions,
+    cluster_options: ClusterOptions,
+    logfile: IO[str],
+):
+    log("Starting Vamb version " + ".".join(map(str, vamb.__version__)), logfile)
+    log("Date and time is " + str(datetime.datetime.now()), logfile, 1)
+    begintime = time.time()
+
+    # Get TNFs, save as npz
+    composition = calc_tnf(comp_options, vamb_options.out_dir, logfile)
+
+    # Parse BAMs, save as npz
+    abundance = calc_rpkm(
+        abundance_options,
+        vamb_options.out_dir,
+        composition.metadata,
+        vamb_options.n_threads,
+        logfile,
+    )
+
+    time_generating_input = round(time.time() - begintime, 2)
+    log(
+        f"\nTNF and coabundances generated in {time_generating_input} seconds.",
+        logfile,
+        1,
+    )
+
+    tnfs, lengths = composition.matrix, composition.metadata.lengths
+    rpkms = abundance.matrix
+
+    if taxonomy_options.taxonomy_predictions_path:
```

Comment:
```suggestion
    if taxonomy_options.taxonomy_predictions_path is not None:
```
to be clear about its possible value